pkg/auth/providers/azure/azure_provider.go

```diff
@@ -400,6 +415,22 @@ func formAzureRedirectURL(config map[string]interface{}) string {
 	)
 }
 
+func extractAnnotations(config map[string]interface{}) map[string]string {
+	result := make(map[string]string)
+	if metadata, ok := config["metadata"].(map[string]interface{}); ok {
+		if annotations, ok := metadata["annotations"].(map[string]interface{}); ok {
+			for k, v := range annotations {
+				if stringValue, ok := v.(string); ok {
+					result[k] = stringValue
+				} else {
+					continue
+				}
+			}
+		}
+	}
+	return result
+}
```

Comment:
Early returns will help flatten things out and make what is going on easier to read. We also don't nee to `else: continue.`
```suggestion
func extractAnnotations(config map[string]interface{}) map[string]string {
	result := make(map[string]string)

	metadata, ok := config["metadata"].(map[string]interface{})
	if !ok {
		return result
	}

	annotations, ok := metadata["annotations"].(map[string]interface{})
	if !ok {
		return result
	}

	for k, v := range annotations {
		if stringValue, ok := v.(string); ok {
			result[k] = stringValue
		}
	}

	return result
}
```

Reply:
Agreed, will refactor.